<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GrapherClass</class>
 <widget class="QMainWindow" name="GrapherClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>950</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>950</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Grapher</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>160</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>160</height>
         </size>
        </property>
        <property name="title">
         <string>Действия над графом</string>
        </property>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>20</y>
           <width>88</width>
           <height>134</height>
          </rect>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QPushButton" name="append_top">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="append_edge">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="delete_element">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="edit_mode">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="PushButtonColor" name="set_color">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="topParameters">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>170</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>170</height>
         </size>
        </property>
        <property name="title">
         <string>Вершина</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Форма</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QComboBox" name="comboFormTop">
           <item>
            <property name="text">
             <string>Круг</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Квадрат</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Ромб</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Треугольник</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Имя</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLineEdit" name="lineNameTop"/>
         </item>
         <item row="5" column="0">
          <widget class="QPushButton" name="deleteAutoEdge">
           <property name="text">
            <string>Удалить петлю</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="edgeParameters">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>200</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>200</height>
         </size>
        </property>
        <property name="title">
         <string>Ребро</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="2" column="0">
          <widget class="QComboBox" name="comboFormEdge">
           <item>
            <property name="text">
             <string>Ненаправленное</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Однонаправленное</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Двунаправленное</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QPushButton" name="revers">
           <property name="text">
            <string>Перенаправить</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Вес </string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Тип</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QSpinBox" name="weightEdge1"/>
         </item>
         <item row="6" column="0">
          <widget class="QSpinBox" name="weightEdge2"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <widget class="Viewer" name="paint_field"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>950</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="action_open"/>
    <addaction name="action_save"/>
    <addaction name="action_savaas"/>
    <addaction name="separator"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Экспорт</string>
    </property>
    <addaction name="action_JPG"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>Правка</string>
    </property>
    <addaction name="action_addtop"/>
    <addaction name="action_addedge"/>
    <addaction name="action_modesee"/>
    <addaction name="action_del"/>
    <addaction name="action_delall"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>Сервис</string>
    </property>
    <addaction name="action_commiy_2"/>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>Помощь</string>
    </property>
    <addaction name="action_help"/>
    <addaction name="separator"/>
    <addaction name="action_aboutProgram"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
   <addaction name="menu_5"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_open">
   <property name="text">
    <string>Открыть</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_save">
   <property name="text">
    <string>Сохранить</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>Выход</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="action_JPG">
   <property name="text">
    <string>Экспорт в &quot;JPG&quot;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="action_del">
   <property name="text">
    <string>Удалить</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Del</string>
   </property>
  </action>
  <action name="action_delall">
   <property name="text">
    <string>Удалить все</string>
   </property>
   <property name="shortcut">
    <string>Shift+Del</string>
   </property>
  </action>
  <action name="action_shortway">
   <property name="text">
    <string>Кратчайший путь</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="action_min">
   <property name="text">
    <string>Минимальный каркас</string>
   </property>
   <property name="shortcut">
    <string>Shift+M</string>
   </property>
  </action>
  <action name="action_elementary">
   <property name="text">
    <string>Множество элементарных циклов</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_savaas">
   <property name="text">
    <string>Сохранить как...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_commiy">
   <property name="text">
    <string>Задача Коммивояжёра</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+K</string>
   </property>
  </action>
  <action name="action_floyd">
   <property name="text">
    <string>Алгоритм Флойда поиска всех кратчайших путей</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+F</string>
   </property>
  </action>
  <action name="action_help">
   <property name="text">
    <string>Справка</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="action_aboutProgram">
   <property name="text">
    <string>О программе</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="action_addtop">
   <property name="text">
    <string>Добавить вершину</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="action_addedge">
   <property name="text">
    <string>Добавить ребро</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="action_modesee">
   <property name="text">
    <string>Режим просмотра и редактирования</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="action_commiy_2">
   <property name="text">
    <string>Задача коммивояжера</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Viewer</class>
   <extends>QGraphicsView</extends>
   <header>viewer.h</header>
  </customwidget>
  <customwidget>
   <class>PushButtonColor</class>
   <extends>QPushButton</extends>
   <header location="global">PushButtonColor.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="grapher.qrc"/>
 </resources>
 <connections/>
</ui>
